presto-hive/src/main/java/io/prestosql/plugin/hive/AcidInfo.java

```diff
@@ -196,13 +276,39 @@ public Builder addDeleteDelta(Path deleteDeltaPath, long minWriteId, long maxWri
             return this;
         }
 
+        public Builder addOriginalFile(Path originalFilePath, long originalFileLength, int bucketId)
+        {
+            requireNonNull(originalFilePath, "originalFilePath is null");
+/*            Path partitionPathFromOriginalPath = originalFilePath.getParent();
+            checkArgument(
+                    partitionLocation.equals(partitionPathFromOriginalPath),
+                    "Partition location in OriginalFile '%s' does not match stored location '%s'",
+                    originalFilePath.getParent().toString(),
+                    partitionLocation);*/
+            bucketIdToOriginalFileInfoMap.put(bucketId, new OriginalFileInfo(originalFilePath.getName(), originalFileLength));
+            return this;
+        }
+
+        public Optional<AcidInfo> buildWithRequiredOriginalFiles(int bucketId)
+        {
+            // 1. Fetch list of all the original files which have same bucket Id
+            // 2. Build AcidInfo
+            checkState(bucketId > -1 && bucketIdToOriginalFileInfoMap.containsKey(bucketId), "Bucket Id to OriginalFileInfo map should have " +
+                    "entry for requested bucket Id");
+            List<DeleteDeltaInfo> deleteDeltas = deleteDeltaInfoBuilder.build();
```

Comment:
As the builder's build method is invoked multiple times, I would not use builder here.

-  make `deleteDeltaInfoBuilder` -> `List<..> deleteDeltaInfos = new ArrayList<>();`
- `ImmutableList.copyOf` here

Reply:
yeah, looks good, fixed.